### DISCUSSION TITLE
Configuration error on using cmake while setting up MeshFix project

### DISCUSSION THREAD

**Discussion by drathin4** [libigl/libigl#1814] [Q&A]
I am trying to setup LibIGL MeshFix (`https://github.com/alecjacobson/meshfix`) but getting a configuration error:

CMake Error at cmake/FindBLAS.cmake:393 (message):
  A required library with BLAS API not found.  Please specify library
  location.
Call Stack (most recent call first):
  CMakeLists.txt:31 (find_package)

-- Configuring incomplete, errors occurred!
See also "C:/Rathin/LibIGL/libigl-meshfix-master/build/CMakeFiles/CMakeOutput.log".

Following were the steps I used while setting it up:
1. I downloaded MeshFix code repository (`root folder`: libigl-meshfix-master).
2. I downloaded LibIGL code repository and placed it under the root folder.
3. I setup LibIGL as usual, issued cmake command with CGAL option, compiled the VisualStudio .sln file. Everything worked correctly till this step.
4. Followed the regular configuration/compilation steps to work on top of LibIGL using MeshFix:
mkdir build (inside root folder)
cd build
cmake -DCMAKE_BUILD_TYPE=Release  ..

The cmake command is giving the error that I have shown above. Kindly let me know how to resolve the error.

**Comment by jdumas** [comment]
Please try to use the official [MeshFix](https://github.com/MarcoAttene/MeshFix-V2.1). It is a more recent version that doesn't have any complicated dependency and works just as well.

**Reply by drathin4** [reply]
I followed the steps you mentioned. But I have a question. I want to use the meshfix function the way it was used earlier: 

`meshfix(V,F,W,G);` (V, W being triangles and F, G being faces)

Also, I didn't find any meshfix.h file for inclusion when I work on top of LibIGL. Can we use the new MeshFix the same way ?

**Comment by alecjacobson** [comment]
My fork of the old meshfix had a "libigl-style wrapper" that you're
probably referring to. I think the old meshfix is "too stale" to revive, so
it'd be better to write a new libigl-style wrapper for the new meshfix. I
haven't done this yet.
-Alec

>

**Reply by drathin4** [reply]
Is that wrapper the `meshfix.h` file ?

**Comment by alecjacobson** [comment]
yes, and the mesh_to_ext_tri_mesh function IIRC

>

**Reply by drathin4** [reply]
Can I use the same `mesh_to_ext_tri_mesh.h` and `ext_tri_mesh_to_mesh.h` files that were used previously to write this new `meshfix.h` wrapper ?

**Comment by alecjacobson** [comment]
haven't tried it. If you do try it and it works or you find a quick fix.
Please let me and I can update it.

>

**Reply by drathin4** [reply]
I have written my own `meshfix.h` wrapper for the MeshFix-V2.1-master project, but haven't tested it yet. I want to know how to integrate this MeshFix project in my existing project on MeshBoolean. I want to execute the `meshfix(V, F, W, G)` command programmatically and not through command line.

Typically while working on top of LibIGL, I keep a fully compiled `libigl` folder and then have the `cmake, build` folders as well as `CMakeLists.txt, main.cpp` files in project directory. Where should I keep the files within `include, src` folders ? I will also have to merge my existing MeshBoolean project's `CMakeLists.txt` file with the MeshFix project's `CMakeLists.txt` file, right ? Could you guide me on the entire process ?

**Reply by jdumas** [reply]
The CMake I wrote for MeshFix was meant to compile MeshFix as a single executable. You need to modify it to compile only the "library" part of MeshFix, without the "main" function. From there it should be pretty simple to include it in your CMake project, following a similar model to the [libigl-example-project](https://github.com/libigl/libigl-example-project/).
